app/screens/edit_server/index.tsx

```diff
@@ -82,14 +101,53 @@ const EditServer = ({closeButtonId, componentId, server, theme}: ServerProps) =>
         }
 
         await DatabaseManager.updateServerDisplayName(server.url, displayName);
+
+        // Update shared password only if it was modified
+        if (preauthSecretModified) {
+            const credentials = await getServerCredentials(server.url);
+            if (credentials) {
```

Comment:
This if is not needed, right? An empty string won't pass through this if, and we want to be able to remove the credentials (by passing an empty string). Or am I missing something?